docs/book/src/topics/aad-integration.md

```diff
@@ -0,0 +1,162 @@
+# AAD Integration
+
+CAPZ can be configured to use Azure Active Directory (AD) for user authentication. In this configuration, you can log into a CAPZ cluster using an Azure AD token. Cluster operators can also configure Kubernetes role-based access control (Kubernetes RBAC) based on a user's identity or directory group membership.
+
+<aside class="note warning">
+
+<h1> Warning </h1>
+
+The following operations to configure Azure AD applications must be completed by an Azure tenant administrator.
+
+</aside>
+
+## Create Azure AD server component
+
+### Create the Azure AD application
+
+```bash
+export CLUSTER_NAME=my-aad-cluster
+```
+
+```bash
+export AZURE_SERVER_APP_ID=$(az ad app create \
+    --display-name "${CLUSTER_NAME}Server" \
+    --identifier-uris "https://${CLUSTER_NAME}Server" \
+    --query appId -o tsv)
+```
+
+### Update the application group membership claims
+```bash
+az ad app update --id ${AZURE_SERVER_APP_ID} --set groupMembershipClaims=All
+```
+
+### Create a service principal for the Azure AD application
+```bash
+az ad sp create --id ${AZURE_SERVER_APP_ID}
+```
+
+The Azure AD service principal needs permissions to perform the following actions:
+
+- Read directory data
+- Sign in and read user profile
+
+```bash
+az ad app permission add \
+    --id ${AZURE_SERVER_APP_ID} \
+    --api 00000003-0000-0000-c000-000000000000 \
+    --api-permissions e1fe6dd8-ba31-4d61-89e7-88639da4683d=Scope 06da0dbc-49e2-44d2-8312-53f166ab848a=Scope 7ab1d382-f21e-4acd-a863-ba3e13f7da61=Role
+az ad app permission grant --id ${AZURE_SERVER_APP_ID} --api 00000003-0000-0000-c000-000000000000
```

Comment:
since your configuration is just oidc and it's not going to query graph api `00000003-0000-0000-c000-000000000000`, you actually don't need this (unless you setup Guard)